### ISSUE TITLE
Questions about optimal models

### ISSUE THREAD

**Issue by WSake** [luozhengrong/HDC-Net#2]
Hello, Mr. Luo, I find some questions to ask you when I repeat your work. 
 1. The optimal epoch mentioned in the paper is 800, and 900 is set in the code, because the code does not set to save the optimal model, but only saves the last model. I would like to ask you how to determine the optimal model for training and test it? Furthermore, is the last model saved by cross-training the loss-optimal model? 
 2. According to the segmentation results, there is a big difference between the ET region and the paper, because there are many segmentation results that are 0. How did you solve this problem at that time? 
 3. Finally, I don't know if you are willing to upload HDC of other structures for reference only. 
 
 Looking forward to your reply, best wishes!

**Comment by luozhengrong** [comment]
Hello, thanks for your question.
1、Choosing 800 or 900 has little impact on the experimental results. Since the learning rate is dynamic learning rate fall so low in the end epoch, the last model and the optimal model are almost the same in segmentation performance. Of course, the 5-fold cross-training method can improve experimental results. For convenience, the code recommend a direct training method. You can utilize cross-validation to get the best experimental results.
2、In the post-processing stage, regions too small for label 4 will be converted to label 1, affecting the ET score. The smaller Batchsize in the experiment also affects it. And using test-time augmentation will improve segmentation performance, but I personally don't like this approach, so ‘use_TTA’ is set to False by default in the inference stage, but you can set it to ‘True’ to improve segmentation accuracy.
3、Recently, we came up with new ideas and optimized the HDC network in order to achieve higher segmentation performance. Future versions may be published as papers.
I hope my answer is helpful

**Comment by luozhengrong** [comment]
> Hello, Mr. Luo, I find some questions to ask you when I repeat your work.
> 
> 1. The optimal epoch mentioned in the paper is 800, and 900 is set in the code, because the code does not set to save the optimal model, but only saves the last model. I would like to ask you how to determine the optimal model for training and test it? Furthermore, is the last model saved by cross-training the loss-optimal model?
> 2. According to the segmentation results, there is a big difference between the ET region and the paper, because there are many segmentation results that are 0. How did you solve this problem at that time?
> 3. Finally, I don't know if you are willing to upload HDC of other structures for reference only.
> 
> Looking forward to your reply, best wishes!

Hello, thanks for your question.
1、Choosing 800 or 900 has little impact on the experimental results. Since the learning rate is dynamic learning rate fall so low in the end epoch, the last model and the optimal model are almost the same in segmentation performance. Of course, the 5-fold cross-training method can improve experimental results. For convenience, the code recommend a direct training method. You can utilize cross-validation to get the best experimental results.
2、In the post-processing stage, regions too small for label 4 will be converted to label 1, affecting the ET score. The smaller Batchsize in the experiment also affects it. And using test-time augmentation will improve segmentation performance, but I personally don't like this approach, so ‘use_TTA’ is set to False by default in the inference stage, but you can set it to ‘True’ to improve segmentation accuracy.
3、Recently, we came up with new ideas and optimized the HDC network in order to achieve higher segmentation performance. Future versions may be published as papers.
I hope my answer is helpful

**Comment by WSake** [comment]
Thank you for your timely reply. I would like to ask if the command entered during the cross verification is the same as that entered during train_all?

**Comment by WSake** [comment]
Thank you for your timely reply. I would like to ask if the command entered during the cross verification is the same as that entered during train_all?



---Original---
From: ***@***.***&gt;
Date: Tue, Mar 29, 2022 10:04 AM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [luozhengrong/HDC-Net] Questions about optimal models (Issue #2)




  
Hello, Mr. Luo, I find some questions to ask you when I repeat your work.
  
The optimal epoch mentioned in the paper is 800, and 900 is set in the code, because the code does not set to save the optimal model, but only saves the last model. I would like to ask you how to determine the optimal model for training and test it? Furthermore, is the last model saved by cross-training the loss-optimal model? 

According to the segmentation results, there is a big difference between the ET region and the paper, because there are many segmentation results that are 0. How did you solve this problem at that time? 

Finally, I don't know if you are willing to upload HDC of other structures for reference only. 
 
Looking forward to your reply, best wishes!
  
Hello, thanks for your question.
 1、Choosing 800 or 900 has little impact on the experimental results. Since the learning rate is dynamic learning rate fall so low in the end epoch, the last model and the optimal model are almost the same in segmentation performance. Of course, the 5-fold cross-training method can improve experimental results. For convenience, the code recommend a direct training method. You can utilize cross-validation to get the best experimental results.
 2、In the post-processing stage, regions too small for label 4 will be converted to label 1, affecting the ET score. The smaller Batchsize in the experiment also affects it. And using test-time augmentation will improve segmentation performance, but I personally don't like this approach, so ‘use_TTA’ is set to False by default in the inference stage, but you can set it to ‘True’ to improve segmentation accuracy.
 3、Recently, we came up with new ideas and optimized the HDC network in order to achieve higher segmentation performance. Future versions may be published as papers.
 I hope my answer is helpful
